### PR TITLE
Fix custom pattern expansion on Windows + comments

### PR DESCRIPTION
**Pull request by msprotz**
https://caml.inria.fr/mantis/view.php?id=7473

### CODE REVIEW

**Comment by msprotz** [comment]
@dra27 thoughts?

File: byterun/win32.c [hunk 1]
```diff
@@ -364,10 +364,15 @@ static void expand_pattern(char * pat)
     return;
   }
   prefix = caml_strdup(pat);
+  /* We need to stop at the first directory or drive boundary, because the
+   * _findata_t structure contains the filename, not the leading directory. */
   for (i = strlen(prefix); i > 0; i--) {
     char c = prefix[i - 1];
     if (c == '\\' || c == '/' || c == ':') { prefix[i] = 0; break; }
   }
+  /* No separator was found, it's a pattern of the form Foo* */
```
Comment (dra27) [inline byterun/win32.c]: Possibly `No separator was found, it's a filename pattern`? For some reason, to me "pattern of the form Foo*" excludes "Foo*Bar", but maybe that's just me!

**Review by dra27 — COMMENTED** [review]
Looks fine - could your test.ml case from the PR form a very quick testsuite test?

**Comment by xavierleroy** [comment]
This PR is an example of what is wrong in our process: the code submitted fixes an old-standing bug in a rarely-exercised part of the system, and is obviously correct.  Yet CI blocks because a Changes entry is missing, reviewer asks for a legitimate but cosmetic change, submitter doesn't react, and the bug remains there.  Who will break the deadlock?

**Comment by dra27** [comment]
@xavierleroy - well, that's part of the Git process in general, surely? However, what will happen here is that if Jonathan doesn't get to it in the next few days, I'll either push a commit here with the Changes file and the test or I'll redo the GPR (former much more likely).

If I'd wanted to unblock the CI, I'd have added the label and restarted test 3 in Travis (takes seconds, even on my iPhone!) but as this one will want a Changes entry, I decided to leave it.

In the background, @avsm and I quietly restarted the failing AppVeyor test for this PR, once we fixed the problem there!

**Comment by gasche** [comment]
Having the Continuous Integration complain about a missing Changes entry is a good thing in the current case, as it really is a mistake to not have a Changes entry here -- it is about fixing a user-observable bug, people in the future may want to know which OCaml versions are affected, eg. to make version choices in Windows-based teaching environments. That seems to be a part of the process working correctly, on the contrary.

(I think Jonathan will break the deadlock on his next context switch. Distributed work incurs latencies, sure, but those at hand here seem reasonable, and well within the bounds of the delay for the next release -- Jonathan did the right thing of submitting a properly-targeted Mantis ticket so there is low risk of forgetting about it.)

**Comment by protz** [comment]
Yeah will submit a new version today... there was an F* release recently 
so I had to context-switch.

**Comment by msprotz** [comment]
For me, this is good to merge.

**Pull request merged**